Description: Don't crash on systems that don't support XF86VM or XRANDR gamma fade
 Test if fade is supported on each screen before calling the fade_setup
 virtual function.
Bug-Ubuntu: https://launchpad.net/bugs/558721
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=618925
Forwarded: https://bugzilla.gnome.org/attachment.cgi?id=161269
Author: Chris Coulson <chris.coulson@canonical.com>

Index: gnome-screensaver-3.4.2/src/gs-fade.c
===================================================================
--- gnome-screensaver-3.4.2.orig/src/gs-fade.c	2012-06-28 10:49:42.915328739 +1200
+++ gnome-screensaver-3.4.2/src/gs-fade.c	2012-06-28 10:49:45.587328645 +1200
@@ -844,6 +844,8 @@
 gs_fade_reset (GSFade *fade)
 {
         int i;
+        struct GSFadeScreenPrivate *screen_priv;
+
         g_return_if_fail (GS_IS_FADE (fade));
 
         gs_debug ("Resetting fade");
@@ -856,8 +858,12 @@
 
         gs_fade_set_alpha (fade, fade->priv->current_alpha);
 
-        for (i = 0; i < fade->priv->num_screens; i++)
-                fade->priv->screen_priv[i].fade_finish (fade, i);
+        for (i = 0; i < fade->priv->num_screens; i++) {
+                screen_priv = &fade->priv->screen_priv[i];
+                if (screen_priv->fade_type != FADE_TYPE_NONE) {
+                        screen_priv->fade_finish (fade, i);
+                }
+        }
 }
 
 static void
